src/syntax/syntax.rkt

```diff
@@ -24,56 +27,97 @@
   [bf (->* () bigvalue?)]
   [fl (->* () value?)]
   [ival (or/c (->* () ival?) #f)]
-  [nonffi (->* () value?)]
-  [->tex string?])
+  [nonffi (->* () value?)])
 
 (define (constant-info constant field) (table-ref constants constant field))
 
 (define-syntax-rule (define-constant constant ctype [key value] ...)
   (table-set! constants 'constant
               (make-hash (list (cons 'type 'ctype) (cons 'key value) ...))))
 
-(define-constant PI real
+(define parametric-constants (make-hash))
+(define parametric-constants-reverse (make-hash))
+
+(define (get-parametric-constant name type)
+  (for/or ([sig (hash-ref parametric-constants name)])
+    (match-define (list* true-name rtype) sig)
+    (and
+     (equal? rtype type)
+     true-name)))
```

Comment:
This is a little ugly. Instead of `list*` I'd use `cons`, since `rtype` is a symbol. And then since I'm iterating over a list of pairs I'd use `in-dict` on the output of `hash-ref`, binding `true-name` and `rtype` in the for loop. Then I'd migrate the `equal?` to a `#:when` clause in the loop and change `for/or` to `for/first` to make it clearer what's going on.

Reply:
Fun fact: this was actually copied from `parametric-operators` ;)